Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -4711,40 +4724,111 @@ def panorama_query_logs_command(args: dict):
+        }
 
-    if result['response']['@status'] == 'error':
-        if 'msg' in result['response'] and 'line' in result['response']['msg']:
-            raise Exception(f"Query logs failed. Reason is: {result['response']['msg']['line']}")
-        else:
-            raise Exception('Query logs failed.')
+        command_results = CommandResults(
+            outputs_prefix='Panorama.Monitor',
+            outputs_key_field='JobID',
+            outputs=query_logs_output,
+            readable_output=tableToMarkdown('Query Logs:',query_logs_output, ['JobID', 'Status', 'Message', 'LogType'], removeNull=True)
+        )
 
-    if 'response' not in result or 'result' not in result['response'] or 'job' not in result['response']['result']:
-        raise Exception('Missing JobID in response.')
+        poll_result = PollResult(
+            response=command_results,
+            continue_to_poll=argToBoolean(args.get('polling', 'false')),
+            args_for_next_run={
+                'query_log_job_id': result.response.result.job,
+                'log-type': log_type,
+                'polling': argToBoolean(args.get('polling', 'false')),
+                'interval_in_seconds': arg_to_number(args.get('interval_in_seconds', 10)),
+                'timeout': arg_to_number(args.get('timeout', 120))
+            },
+            partial_result=CommandResults(
+                readable_output=f"Fetching logs for job ID {result.response.result.job}..."
+            )
+        )
 
-    query_logs_output = {
-        'JobID': result['response']['result']['job'],
-        'Status': 'Pending',
-        'LogType': log_type,
-        'Message': result['response']['result']['msg']['line']
-    }
+    else:
+        # Only used in subsequent polling executions
 
-    return_results({
-        'Type': entryTypes['note'],
-        'ContentsFormat': formats['json'],
-        'Contents': result,
-        'ReadableContentsFormat': formats['markdown'],
-        'HumanReadable': tableToMarkdown('Query Logs:', query_logs_output, ['JobID', 'Status'], removeNull=True),
-        'EntryContext': {"Panorama.Monitor(val.JobID == obj.JobID)": query_logs_output}
-    })
+        res = panorama_get_traffic_logs(job_id)
+        parsed: SimpleNamespace = parse_response(res)
+        if parsed.response.status == 'error':
+            if parsed.response.result.msg.line:
+                raise Exception(
+                    f'Query logs failed. Reason is: {parsed.response.result.msg.line}'
+                )
+            else:
+                raise Exception(
+                    f'Query logs failed.'
+                )
+        
+        if not parsed.response.result.job.id:
+            raise Exception('Missing JobID status in response.')
+
+        query_traffic_logs_output = {
+            'JobID': job_id,
+            'Status': 'Pending',
+            'LogType': log_type
+        }
+        if parsed.response.result.job.status == 'FIN':
```

Comment:
```suggestion
        if parsed.response.result.job.status.upper() == 'FIN':
```

to be on the safe side